moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -326,7 +346,15 @@ const ServoParameters& Servo::getParameters() const
 
 sensor_msgs::JointStateConstPtr Servo::getLatestJointState() const
 {
-  return joint_state_subscriber_->getLatest();
+  auto msg = moveit::util::make_shared_from_pool<sensor_msgs::JointState>();
+  auto current_state_time = planning_scene_monitor_->getStateMonitor()->getCurrentStateAndTime();
+
+  msg->header.stamp = current_state_time.second;
+  msg->name = current_state_time.first->getJointModelGroup(parameters_.move_group_name)->getActiveJointModelNames();
+  current_state_time.first->copyJointGroupPositions(parameters_.move_group_name, msg->position);
+  current_state_time.first->copyJointGroupPositions(parameters_.move_group_name, msg->velocity);
+
+  return msg;
```

Comment:
This is now a convenience function that retains the original functionality but should probably not be used.  The reason for that is it does a bunch of unnecessary work if all you wanted was the latest joint values from the current robot state.  In that case, whatever is calling this should instead just use the CSM through the PSM it already has.

I did this because I didn't want to brake any code that depends on this API.  I think if this change is adopted into MoveIt2 this API function should just be removed.

Reply:
Since this function isn't used anywhere, I think you should delete it.

Reply:
I removed the method and added a `MIGRATION.md` to moveit_servo with a note about how to get the latest joint positions using the PSM.